homeassistant/components/ecobee/sensor.py

```diff
@@ -165,6 +166,6 @@ async def async_update(self):
             for item in sensor["capability"]:
                 if item["type"] != self.entity_description.key:
                     continue
-                thermostat = self.data.ecobee.get_thermostat(self.index)
+                thermostat = self.thermostat(self.index)
```

Comment:
To be clear what I was thinking was 

```python
self.thermostat = self.data.ecobee.get_thermostat(self.index)
self._state = self.thermostat["runtime"]
```

seems redundant to create a local variable now that we already have self.thermostat. This could be shortened down more, but I figured `self.data.ecobee.get_thermostat(self.index)` was needed to update the values?

Reply:
Let me know if I am missing something

Reply:
self.thermostat should work.  Let me test.  It was acting funky without the local variables

Reply:
Ok. It's working.  I think those were all the issues.

Reply:
Nope, it broke it. We can’t cache the thermostat.